swupd/swupd_test.go

```diff
@@ -191,6 +190,82 @@ func TestFullRunDelta(t *testing.T) {
 	// not done by new swupd since it seems the client doesn't take advantage of them.
 }
 
+// Test an import cycle for includes subtraction
+func TestRunBundleCycle(t *testing.T) {
+	ts := newTestSwupd(t, "bundle-cycle-")
+	defer ts.cleanup()
+
+	// Version 10.
+	ts.Bundles = []string{"os-core", "test-bundle1", "test-bundle2"}
+
+	ts.mkdir("image/10/os-core/a")
+	ts.write("image/10/test-bundle1/a/b/t1", "t1")
+	ts.write("image/10/test-bundle2/a/b/t2", "t2")
+	ts.write("image/10/noship/test-bundle1-includes", "test-bundle2")
+	ts.write("image/10/noship/test-bundle2-includes", "test-bundle1")
+
+	ts.createManifestsFromChroots(10)
+	ts.createFullfiles(10)
+
+	ts.createPack("os-core", 0, 10, ts.path("image"))
+	info1 := ts.createPack("test-bundle1", 0, 10, ts.path("image"))
+	info2 := ts.createPack("test-bundle2", 0, 10, ts.path("image"))
+	mustHaveFullfileCount(t, info1, 3) // b dir, t1 file and the test-bundle1 file.
+	mustHaveFullfileCount(t, info2, 3) // b dir, t2 file and the test-bundle2 file.
+
+	mustValidateZeroPack(t, ts.path("www/10/Manifest.os-core"), ts.path("www/10/pack-os-core-from-0.tar"))
+	mustValidateZeroPack(t, ts.path("www/10/Manifest.test-bundle1"), ts.path("www/10/pack-test-bundle1-from-0.tar"))
+	mustValidateZeroPack(t, ts.path("www/10/Manifest.test-bundle2"), ts.path("www/10/pack-test-bundle2-from-0.tar"))
+
+	osCore := ts.parseManifest(10, "os-core")
+	testBundle1 := ts.parseManifest(10, "test-bundle1")
+	testBundle2 := ts.parseManifest(10, "test-bundle2")
+	checkIncludes(t, testBundle1, "os-core", "test-bundle2")
+	checkIncludes(t, testBundle2, "os-core", "test-bundle1")
+	checkFileInManifest(t, osCore, 10, "/a")
+	checkFileInManifest(t, testBundle1, 10, "/a/b")
```

Comment:
Do we need something like `checkFileNotInManifest(t, testBundle1, 10, "/a")` and for `testBundle2`?

Reply:
Yea, seems pretty reasonable to add, though I'll need to create `checkFileNotInManifest`.

Reply:
Looks like `fileNotInManifest` was there and did what I wanted for this.